### PR TITLE
docs: Make the generated README more contributor oriented

### PR DESCRIPTION
**Pull request by hillairet**
The current state of the README is quite hard to follow for people not familiar with the setup already.

One overall change was in the section titles and levels
* I used "How to ..." because it makes the purpose of the section clearer for the writer and the reader in my opinion.
* I adjusted the section levels because it was really not consistent.

Two very confusing sections:
1. The initialization of the github repo feels out of place because it happens once and is done by the person running scaf anyway. This should be in scaf itself, not in the generated README so I removed it
2. The management of secrets seemed to be jumping between sealedSecrets and .envrc file so I made two clean sections instead

To make the setup clearer for contributors I made two sections:

* one for setting up the environment, done once
* one for spinning up the app with commands that must be run every time

### CODE REVIEW

**Comment by daveoconnor** [comment]
That looks better, adding my notes as well as some comments on those based on your changes, from my own thinking when I was going through the older process, some of which still applies:

> When the developer has a new project/codebase set up the docs are orientated towards the users who will come onto the project, rather than the user who needs to set up the project.  

Maybe less the case now with these changes though maybe worth a second look all the same with that in mind and the idea later for maybe two sets of docs.

> ~~E.g. The docs say the user has to install Kubectl, Kind, and Tilt, but that user already has those set up. Not bad, but still a double take moment for me.~~ Then the user goes on reading the "Initial Project Setup" instructions, which say to run `make compile` which fails because of 1password secrets

This would probably mean we'd want to move the .envrc step earlier in the doc with your change.

> Thoughts: This needs refinement? 
> At a minimum adding 1password instructions would be better before `make compile`, but maybe there should be two `README.md`, one pared back one for the initial setup which most developers will look for of habit, and a second like `README_CONTRIBUTORS.md`, with the first README having an instruction the follows the rest of the setup to run `mv README_CONTRIBUTORS.md README.md`.

> Distinct instructions might make them easier to change when updating the docs too.

> Alternatively, a `README.md`with two distinct sections for setup/contributors.

Just some thoughts.

**Comment by hillairet** [comment]
I don't understand the last part of your comment:
In which situation are the steps of running scaf and setting up the generated project with `make compile` done by two people at two different times?
For me (and I could be totally wrong), whoever runs scaf will also setup (once!) the generated project all the way to where it can be handed off to any contributor, Sixie or not.
Therefore the project setup docs should be in the scaf root README and anything related to maintaining and developing the generated project belongs in the project's README.

Maybe we're on the same page and I misunderstood your last comments!  :sweat_smile:

File: {{cookiecutter.project_slug}}/README.md [hunk 1]
```diff
@@ -1,70 +1,63 @@
...
-To prepare your development environment you can use the following commands:
+To stop the local kubernetes cluster once you are done working:
+
+       $ kind delete cluster --name {{ cookiecutter.project_slug }}
+
+
+### Update dependencies
+
+To update the backend app dependencies, you must edit the `backend/requirements/*.txt` files.
+Once you have made your changes, you need to regenerate the `backend/requirements/*.txt` files you must run:
```
Comment (rochecompaan) [inline {{cookiecutter.project_slug}}/README.md:57]: That is incorrect; you must modify the `.in` files in the requirements directory. The `*.txt` files are generated once you run `make compile`.

**Review by rochecompaan — CHANGES_REQUESTED** [review]
Nice! We need a few small updates.

**Comment by hillairet** [comment]
It would be good to merge sooner rather than later to avoid more conflicts and losing stuff.

**Review by rochecompaan — APPROVED** [review]

**Comment by hillairet** [comment]
Merging with the idea that it's good enough to be in and might need to be improved in a later PR.

File: {{cookiecutter.project_slug}}/README.md [hunk 2]
```diff
@@ -75,7 +79,7 @@ Django Admin that will be accesible at [http://localhost:8009/admin](http://loca
 
 To create a superuser use the following commands:
 
-    $ make shell
+    $ make shell-backend
```
Comment (daveoconnor) [inline {{cookiecutter.project_slug}}/README.md:82]: with a newly generated project it seems like `make shell-backend` no longer exists, and `make shell` doesn't work.

Comment (daveoconnor) [reply {{cookiecutter.project_slug}}/README.md:82]: I've reopened a ticket about that https://github.com/sixfeetup/scaf/issues/236
